agent/engine/docker_client_test.go

```diff
@@ -438,20 +440,24 @@ func TestStartContainerTimeout(t *testing.T) {
 	mockDocker, client, _, _, _, done := dockerClientSetup(t)
 	defer done()
 
-	testDone := make(chan struct{})
 	wait := &sync.WaitGroup{}
 	wait.Add(1)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockDone := make(chan struct{})
+	mockContext := NewMockContext(mockCtrl)
+	mockContext.EXPECT().Done().Return(mockDone).AnyTimes()
```

Comment:
Why those are `AnyTimes`, shouldn't this be called just once or twice?

Reply:
I think it should be called twice, because the `CreateContainer` will also call `InspectContainer` where a new child context will be generated, but test using mock context will be deleted anyway.

Reply:
My opinion: The number of times context is invoked is an implementation detail and not part of tested function's contract.  IE, its not something we would want the test to fail for.